docs: Add comprehensive Files API and Vector Store integration doc (#3279)
llamastack/llama-stack: docs/source/concepts/apis.md

- **VectorIO**: perform operations on vector stores, such as adding documents, searching, and deleting documents
- **Files**: manage file uploads, storage, and retrieval
- **Telemetry**: collect telemetry data from the system
- **Post Training**: fine-tune a model
- **Tool Runtime**: interact with various tools and protocols
- **Responses**: generate responses from an LLM

We are working on adding a few more APIs to complete the application lifecycle. These will include:
- **Batch Inference**: run inference on a dataset of inputs
- **Batch Agents**: run agents on a dataset of inputs
- **Synthetic Data Generation**: generate synthetic data for model development
- **Batches**: OpenAI-compatible batch management for inference

## File Operations and Vector Store Integration

The Files API and Vector Store APIs work together through file operations, enabling automatic document processing and search. This integration implements the [OpenAI Vector Store Files API specification](https://platform.openai.com/docs/api-reference/vector-stores-files) and allows you to:

- Upload documents through the Files API
- Automatically process and chunk documents into searchable vectors

> - Store processed content in vector databases (FAISS, SQLite-vec, Milvus, etc.)

Reviewer: ```suggestion
- Store processed content in vector databases based on the availability of our providers
```
I'd probably hyperlink providers to the vector DB providers

Author: done,
link added to ../providers/
